lib/web_ui/lib/src/engine/app_bootstrap.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:js/js.dart';
+
+import 'js_interop/js_loader.dart';
+import 'js_interop/js_promise.dart';
+
+/// A class that controls the coarse lifecycle of a Flutter app.
+class AppBootstrap {
+  /// Construct a FlutterLoader
+  AppBootstrap({Function? initEngine, Function? runApp, Function? cleanApp}) :
+    _initEngine = initEngine, _runApp = runApp, _cleanApp = cleanApp;
+
+  // TODO(dit): Be more strict with the below typedefs, so we can add incoming params for each function.
+
+  // A function to initialize the engine
+  final Function? _initEngine;
+
+  // A function to run the app
+  final Function? _runApp;
+
+  // A function to clear the app (optional)
+  final Function? _cleanApp;
```

Comment:
Let's remove `_cleanApp`. Instead, when/if we are implementing the tear-down functionality we'll probably use `Promise`-based API to communicate to the developer that Flutter-side clean-up is finished.

Reply:
Removed. For now the FlutterApp object is empty :)